modules/magnetometer/src/main/native/android/dalvik/DalvikMagnetometerService.java

```diff
@@ -57,35 +57,19 @@
     public DalvikMagnetometerService(Activity activity) {
         debug = Util.isDebug();
         sensorManager = (SensorManager) activity.getSystemService(SENSOR_SERVICE);
-        Util.setLifecycleEventHandler(new LifecycleEventHandler() {
-            @Override
-            public void lifecycleEvent(String event) {
-                if (event != null && !event.isEmpty()) {
-                    switch (event) {
-                        case "pause":
-                            unregisterListener();
-                            break;
-                        case "resume":
-                            registerListener();
-                            break;
-                        default:
-                            break;
-                    }
-                }
-            }
-        });
-        registerListener();
     }
 
-    private void setup(int frequency) {
+    private void start(int frequency) {
         if (debug) {
-            Log.v(TAG, "DalvikMagnetometerService::setup");
-        }
-        if (this.frequency != frequency) {
-            this.frequency = frequency;
-            unregisterListener();
-            registerListener();
+            Log.v(TAG, "DalvikMagnetometerService::start");
         }
+        this.frequency = frequency;
+        unregisterListener();
+        registerListener();
+    }
+    
+    private void stop() {
+        unregisterListener();
     }
 
     private void registerListener() {
```

Comment:
I've been revisiting the `sensorManager.registerListener` API, and it uses `microseconds` for the rate.

This change is required:
```
 double rateInSeconds = frequency > 0 ? (1.0d / (double) frequency) : 0.1;
 int rateInMicroseconds = (int) (rateInSeconds * 1_000_000);
            
sensorManager.registerListener(this,
                        sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER),
                        rateInMicroseconds);

sensorManager.registerListener(this,
                        sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD),
                        rateInMicroseconds);
```